### PR TITLE
Enhance uninstall action

### PR DESCRIPTION
**Pull request by ckotzbauer**
Uninstall enhancements:
- Allow lookahead regex for "release_filter" (helm -f only allows perl-compatible regex which doesn't support lookaheads)
- Add date-filtering to achieve "max-days since last upgrade" use-case
- Make Branch-Property of Helm-Release configurable (defaults to .branch)

Enhance Docs

### CODE REVIEW

**Review by agentschmitt — DISMISSED** [review]

**Review by devRoemer — APPROVED** [review]

File: uninstall/action.yml [hunk 1]
```diff
@@ -40,37 +47,60 @@ runs:
...
+            COMPARE_DATE=$(date -d "now - ${MAX_AGE_IN_DAYS} days" +%s)
+
+            if [ "${LAST_UPGRADE_DATE}" -le "${COMPARE_DATE}" ]; then
+              echo "$RELEASE"
+              echo -e "$RELEASE" >> /tmp/HELM_RELEASES
+            fi
+          done
+        fi
+
+        echo -n "${HELM_RELEASES}" > /tmp/HELM_RELEASES
```
Comment (agentschmitt) [inline uninstall/action.yml]: this will  overwrite your  releases filtered by max-age, so this filter will never work

File: uninstall/action.yml [hunk 2]
```diff
@@ -40,37 +47,60 @@ runs:
...
+          echo -n > /tmp/HELM_RELEASES
+          for RELEASE in $HELM_RELEASES
+          do
+            LAST_UPGRADE=$(echo "${HELM_RELEASES_JSON}" | jq -r ". | map(select(.name == \"$RELEASE\"))[0].updated" | cut -d ' ' -f1-2)
+            LAST_UPGRADE_DATE=$(date -d "${LAST_UPGRADE}" +%s)
+            COMPARE_DATE=$(date -d "now - ${MAX_AGE_IN_DAYS} days" +%s)
+
+            if [ "${LAST_UPGRADE_DATE}" -le "${COMPARE_DATE}" ]; then
+              echo "$RELEASE"
+              echo -e "$RELEASE" >> /tmp/HELM_RELEASES
```
Comment (agentschmitt) [inline uninstall/action.yml:97]: this result will be overwritten

**Review by agentschmitt — COMMENTED** [review]
@ckotzbauer i think i found a bug in your workflow, please have a look.

**Comment by ckotzbauer** [comment]
@agentschmitt Thanks! I removed the last output to the file, as it was written before already.

**Review by agentschmitt — APPROVED** [review]

**Pull request merged**